src/Widgets/AddressWidget/AddressWidgetComponent.js

```diff
@@ -66,3 +33,84 @@ const Logo = Scrivito.connect(() => {
     </Scrivito.LinkTag>
   );
 });
+
+const Address = Scrivito.connect(({ addressWidget }) => {
+  const localityRegionPostalCode = [
+    addressWidget.get('locationLocality'),
+    addressWidget.get('locationRegion'),
+    addressWidget.get('locationPostalCode'),
+  ].filter(n => n).join(' ');
+
+  const lines = [
+    addressWidget.get('locationName'),
+    addressWidget.get('locationStreetAddress'),
+    localityRegionPostalCode,
+    addressWidget.get('locationCountry'),
+  ].filter(n => n);
+
+
+  if (!lines.length) {
+    return (
+      <InPlaceEditingPlaceholder>
+        Provide the location in the address widget properties.
+      </InPlaceEditingPlaceholder>
+    );
+  }
+
+  const breakedLines = lines.reduce((array, value) => {
+    array.push(value);
+    array.push(<br key={ value } />);
+    return array;
+  }, []);
+
+  return <p>{ breakedLines }</p>;
+});
+
+function Table(values) {
+  const columns = Object.keys(values).map(name => {
+    return <TableColumn key={ name } name={ name } value={ values[name] } />;
+  });
+
+  if (columns.length === 0) {
```

Comment:
This does not work, since all four props are always given (but with empty values).

Reply:
What do you think about the following? It also uses the newer `Object.entries` method of es7:

```javascript
function Table(props) {
  const lines = Object.entries(props).filter(([_prop, value]) => value);

  if (!lines.length) {
    return null;
  }

  return (
    <table>
      <tbody>
        {
          lines.map(([name, value]) =>
            <TableColumn key={ name } name={ name } value={ value } />)
        }
      </tbody>
    </table>
  );
}
```